src/system.rs

```diff
@@ -164,21 +210,28 @@ pub type TypeId = usize;
 /// Defines a cache that can be used to store resolved [Type]s of a [Schema]
 #[derive(Debug, Clone)]
 pub struct TypeStore {
+    imports: HashMap<String, TypeId>, // stores all the imported types of a schema
```

Comment:
`import` is becoming rather overloaded; sometimes it's the grammatical element and sometimes it's a type that was imported. I think this is the latter; if that's the case, it might help to rename this `imported_types` (or `imported_type_ids_by_name` if you want to be verbose but consistent).

Reply:
+1 especially to `imported_type_ids_by_name`